app/api/webhook/github/route.ts

```diff
@@ -255,96 +251,55 @@ export const POST = async (req: Request) => {
 
 	await incrementUsage(prisma, owner)
 
+	const {issue, pull_request: pr} = payload
+	const prComment = issue?.pull_request
+
 	/**
 	 * Repo commands
 	 */
 	try {
-		const queryRes: {
-			repository: {
-				description?: string
-			}
-		} = await octokit.graphql(
-			`
-        query Repo($name: String!, $owner: String!) {
-          repository(name: $name, owner: $owner) {
-            description
-          }
-        }
-      `,
-			{
-				name,
-				owner
-			}
-		)
-
-		if (!queryRes?.repository)
-			return new Response('Could not get repo', {status: 401})
-
-		const labelsRes: {
-			repository: {
-				description?: string
-				labels: {
-					nodes: Label[]
-				}
-			}
-		} = await octokit.graphql(
-			`
-        query Labels($name: String!, $owner: String!) {
-          repository(name: $name, owner: $owner) {
-            labels(first: 100) {
-              nodes {
-                id
-                name
-                description
-              }
-            }
-          }
-        }
-      `,
-			{
-				name,
-				owner
-			}
-		)
-
-		if (!labelsRes?.repository?.labels?.nodes)
-			throw new Error('Could not get labels')
-
-		const allLabels: Label[] = labelsRes.repository.labels.nodes
-
-		console.log(`Comment by ${comment?.author_association} in ${owner}/${name}`)
-
-		const {description: repoDescription} = queryRes.repository
+		const {labels: allLabels, description: repoDescription} = await getRepoMeta({
+			octokit,
+			owner,
+			name
+		})
 
 		const isComment = action === 'created'
+		const labels = issue?.existingLabels?.map((l: Label) => l.name).join(', ')
 
-		// Note: issue number has been omitted
-		const engPrompt = `
-Hey, here's an incoming ${isComment ? 'comment' : 'issue'}.
+		const prompt = `
+Hey, here's an incoming ${isComment ? 'comment' : pr ? 'PR' : 'issue'}.
 First, some context:
-Repo owner: ${owner}.
-Repo name: ${name}.
+Repo full name: ${owner}/${name}.
 Repo description: ${repoDescription}.
-All repo labels: ${allLabels
-			.map(
-				({name, description}) => `${name}: ${description?.replaceAll(';', ',')}`
-			)
-			.join('; ')}.
-Issue ID: ${issueId}.
-Issue number: ${issueNumber}.
-Issue title: ${title}.
-Issue body: ${body}.
-Issue labels: ${existingLabelNames.join(', ')}.
-Your instructions: ${instructions}.
-${isComment ? `Comment by @${comment.user.login}: ${comment?.body}.` : ''}
+${
+	pr || prComment
+		? `
+PR number: ${pr?.number || issue.number}.
+PR title: ${pr?.title || issue.title}.
+PR body: ${pr?.body || issue.body}.
+	`
+		: `
+Issue number: ${issue.number}.
```

Comment:
Instead of passing the issue number to maige, what if we just input it directly into the tool when we configure the agent?

Reply:
Ok, `issueNumber` is passed directly to `engineer()` and `issueId` is passed to `labelTool()`.

The only remaining tool that might use `issueNumber` is `githubTool()` which uses it in an open-ended way. Only idea I have for replacing it is to add to the prompt `"write ISSUE_NUMBER any time you want to reference the issue number"` then `output.replaceAll("ISSUE_NUMBER", issueNumber)` but this still requires the LLM to be precise. In need of a long-term solution for this pattern.

Reply:
Does githubTool just accept any random json from the LLM?